samples/doc-snippets/src/main/java/com/example/application/RemoteMcpWeatherAgent.java

```diff
@@ -0,0 +1,41 @@
+package com.example.application;
+
+import akka.http.javadsl.model.headers.Authorization;
+import akka.http.javadsl.model.headers.OAuth2BearerToken;
+import akka.javasdk.agent.Agent;
+import akka.javasdk.agent.RemoteMcpTools;
+import akka.javasdk.annotations.ComponentId;
+
+import java.util.Set;
+
+@ComponentId("remote-mcp-weather-agent")
+public class RemoteMcpWeatherAgent extends Agent {
+
+  record AgentResponse(){}
+  private static final String SYSTEM_MESSAGE =
+    """
+    You are a weather agent.
+    Your job is to provide weather information.
+    You provide current weather, forecasts, and other related information.
+
+    The responses from the weather services are in json format. You need to digest it into human language. Be aware that
+    Celsius temperature is in temp_c field. Fahrenheit temperature is in temp_f field.
+    """.stripIndent();
+
+  // tag::mcp-function-tool[]
+  public Effect<AgentResponse> query(String message) {
+    return effects()
+        .systemMessage(SYSTEM_MESSAGE)
+        .mcpTools(
+            RemoteMcpTools.fromServer("http://weather-service/mcp"), // <1>
```

Comment:
does it still work if https is used, by accident?

Reply:
Not for local service calls no, that's always encrypted/decrypted by service mesh. We could perhaps add overloads with `.fromService(name)` `.fromService(name, path)`.